src/execution_tree/primitives/div_operation.cpp

```diff
@@ -27,321 +29,380 @@ HPX_REGISTER_DERIVED_COMPONENT_FACTORY(
-                                "of dimensions");
+                                                "div_operation::eval",
+                                                "the div_operation primitive requires that the "
+                                                        "arguments given by the operands array are valid");
                         }
-                    }),
-                    detail::map_operands(operands, numeric_operand, args)
-                );
+
+                        auto this_ = this->shared_from_this();
+                        return hpx::dataflow(hpx::util::unwrapping(
+                                [this_](operands_type && ops) -> primitive_result_type
+                                {
+                                    std::size_t lhs_dims = ops[0].num_dimensions();
+                                    switch (lhs_dims)
+                                    {
+                                        case 0:
+                                            return this_->div0d(std::move(ops));
+
+                                        case 1:
+                                            return this_->div1d(std::move(ops));
+
+                                        case 2:
+                                            return this_->div2d(std::move(ops));
+
+                                        default:
+                                            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                                                                "div_operation::eval",
+                                                                "left hand side operand has unsupported number "
+                                                                        "of dimensions");
+                                    }
+                                }),
+                                             detail::map_operands(operands, numeric_operand, args)
+                        );
+                    }
+                };
             }
-        };
-    }
 
-    // implement '/' for all possible combinations of lhs and rhs
-    hpx::future<primitive_result_type> div_operation::eval(
-        std::vector<primitive_argument_type> const& args) const
-    {
-        if (operands_.empty())
-        {
-            return std::make_shared<detail::div>()->eval(args, noargs);
-        }
+            // implement '/' for all possible combinations of lhs and rhs
+            hpx::future<primitive_result_type> div_operation::eval(
+                    std::vector<primitive_argument_type> const& args) const
+            {
+                if (operands_.empty())
+                {
+                    return std::make_shared<detail::div>()->eval(args, noargs);
+                }
 
-        return std::make_shared<detail::div>()->eval(operands_, args);
-    }
-}}}
+                return std::make_shared<detail::div>()->eval(operands_, args);
+            }
+        }}}
```

Comment:
All of this file was reformatted, please try to avoid that. Also the file misses an EOL character at its end.